docs/rebase.md

```diff
@@ -157,3 +157,148 @@ git commit -m "update buildfiles"
+Can serve as an example of what arguments `rebase.py` expects.
+
+### rebase.sh
+
+Primary consumer of release image references.
+Script responsible for updating MicroShift's manifests, image references, and go.mod references based on contents of release images.
+It's executed by `rebase.py` to capture output and exit code.
+
+It also creates a git branch which name consists of prefix `rebase-`, version and stream (e.g. `4.13.0-0.nightly`), and creation datetimes of AMD and ARM nightly releases (e.g. `_amd64-2023-01-31-072358_arm64-2023-01-31-233557`).
+
+### rebase.py
+
+Python script responsible for interacting with remote `openshift/microshift` repository and communicating with GitHub API. 
+Uses GH App's ID and key to generate IAT (Installation Access Token) which is like Personal Access Token except its lifetime is 1 hour.
+
+It also executes `rebase.sh` to capture its output and exit code.
+In case of error, it will save `rebase.sh` output to a file, commit it together with any modified files, push changes, and create a PR to inform about the failure in visible way.
+
+`rebase.py` interacts with remote git repository:
+- Creates new, temporary `git remote`: `https://x-access-token:{token}@github.com/openshift/microshift`.
+- Fetches remote references (branches) to check if rebase branch already exists, compare it with local, and decide if it needs update.
+- (Force) pushes local branch to `openshift/microshift`
+
+`rebase.py` interacts with GitHub API:
+- Lists existing PRs to find one matching local branch name, so it'll force push changes and comment under PR that it was refreshed (can happen if no new nightlies were produced, but another PR was merged on microshift repository, so what happens is job is rebasing the rebase PR on newer main branch)
+- Creates a PR if needed.
+- Posts a PR comment which extra information that is important, but shouldn't be part of PR's description.
+- Cleans up branches of closed PRs (branches of merges PRs are automatically deleted, otherwise we need to clean the up).
+
+
+### Testing changes to rebase automation
+
+To reduce a need of Pull Request synchronization between repositories, keep logic together, and allow for easier testing and developing, all scripts related to rebasing are residing in `scripts/auto-rebase/` directory in this repository.
+
+[Job's logic is to only execute](https://github.com/openshift/release/blob/1976df80259f0357c0842700a70a64433988892a/ci-operator/step-registry/openshift/microshift/rebase/openshift-microshift-rebase-commands.sh)
+`scripts/auto-rebase/rebase_job_entrypoint.sh` which gathers necessary arguments and passes them to `rebase.py` and is implemented in a way expecting to be executed in the job's container but can be a guidance on what arguments provide to `rebase.py` locally.
+
+
+#### Getting release image build references
+
+Rebase procedure expects references to AMD64 and ARM64 release images.
```

Comment:
It needs the CSI image, too, now that #1108 landed, right?

Reply:
That is correct

Reply:
LVMS added